src/include/kernel/tokens.h

```diff
@@ -85,6 +85,7 @@
 #define SOF_TKN_INTEL_SSP_QUIRKS		504
 #define SOF_TKN_INTEL_SSP_TDM_PADDING_PER_SLOT	505
 #define SOF_TKN_INTEL_SSP_BCLK_DELAY		506
+#define SOF_TKN_INTEL_SSP_CLK_EARLY_START	507
 
```

Comment:
Fail to cherry-pick. maybe we should merge this patch with the 'ipc: dai-intel: add clk_early_start flag'?

No codespell typos will be found - file '/usr/share/codespell/dictionary.txt': No such file or directory
WARNING: Please update ABI in accordance with http://semver.org
#9: FILE: src/include/uapi/user/tokens.h:99:
 #define SOF_TKN_INTEL_SSP_TDM_PADDING_PER_SLOT 505
+#define SOF_TKN_INTEL_SSP_CLK_EARLY_START      507

total: 0 errors, 1 warnings, 33 lines checked

Reply:
> Fail to cherry-pick but the ABI is already updated in 'ipc: dai-intel: add clk_early_start flag'...
> 
> No codespell typos will be found - file '/usr/share/codespell/dictionary.txt': No such file or directory
> WARNING: Please update ABI in accordance with http://semver.org
> #9: FILE: src/include/uapi/user/tokens.h:99:
> #define SOF_TKN_INTEL_SSP_TDM_PADDING_PER_SLOT 505
> +#define SOF_TKN_INTEL_SSP_CLK_EARLY_START 507
> 
> total: 0 errors, 1 warnings, 33 lines checked

Please cherry-pick from 2ce9552 to 806e766, thanks.

Reply:
Does this token mean enabling all clocks i.e. MCLK, BCK and FRAME or just BCLK and FRAME ?
Also where do we set the duration of the "EARLY_START" ? or are we are meaning to enable the BCLK and FRAME as soon as we configure the DAI ? or at any other stage during PCM ops ?

Reply:
> Does this token mean enabling all clocks i.e. MCLK, BCK and FRAME or just BCLK and FRAME ?
> Also where do we set the duration of the "EARLY_START" ? or are we are meaning to enable the BCLK and FRAME as soon as we configure the DAI ? or at any other stage during PCM ops ?

It means MCLK and BCLK. i.e. Doesn't affect FRAME. With this token enabled, MCLK/BCLK will start during hw_parames and stop during hw_free.